test/parallel/test-dgram-multicast-set-interface-lo.js

```diff
@@ -0,0 +1,293 @@
+        console.error('[PARENT] All workers have died.');
+        console.error('[PARENT] Fail');
+
+        killChildren(workers);
+
+        process.exit(1);
+      }
+    });
+
+    worker.on('message', function(msg) {
+      if (msg.listening) {
+        listening += 1;
+
+        if (listening === listeners) {
+          // All child process are listening, so start sending.
+          sendSocket.sendNext();
+        }
+      } else if (msg.message) {
+        worker.messagesReceived.push(msg.message);
+
+        if (worker.messagesReceived.length === worker.messagesNeeded.length) {
+          done += 1;
+          worker.isDone = true;
+          console.error('[PARENT] %d received %d messages total.',
+                        worker.pid,
+                        worker.messagesReceived.length);
+        }
+
+        if (done === listeners) {
+          console.error('[PARENT] All workers have received the ' +
+                        'required number of ' +
+                        'messages. Will now compare.');
+
+          Object.keys(workers).forEach(function(pid) {
+            const worker = workers[pid];
+
+            let count = 0;
+
+            worker.messagesReceived.forEach(function(buf) {
+              for (let i = 0; i < worker.messagesNeeded.length; ++i) {
+                if (buf.toString() === worker.messagesNeeded[i]) {
+                  count++;
+                  break;
+                }
+              }
+            });
+
+            console.error('[PARENT] %d received %d matching messages.',
+                          worker.pid,
+                          count);
+
+            assert.strictEqual(count, worker.messagesNeeded.length,
+                               'A worker received ' +
+                               'an invalid multicast message');
+          });
+
+          clearTimeout(timer);
+          console.error('[PARENT] Success');
+          killChildren(workers);
+        }
+      }
+    });
+  }
+
+  const sendSocket = dgram.createSocket({
+    type: UDP[FAM],
+    reuseAddr: true
+  });
+
+  // Don't bind the address explicitly when sending and start with
+  // the OSes default multicast interface selection.
+  sendSocket.bind(common.PORT, ANY[FAM]);
+  sendSocket.on('listening', function() {
+    console.error(`outgoing iface ${interfaceAddress}`);
+  });
+
+  sendSocket.on('close', function() {
+    console.error('[PARENT] sendSocket closed');
+  });
+
+  sendSocket.sendNext = function() {
+    const msg = messages[i++];
+
+    if (!msg) {
+      sendSocket.close();
+      return;
+    }
+    console.error(TMPL(NOW, msg.tail));
+    const buf = Buffer.from(TMPL(msg.tail));
+    if (msg.newAddr) {
+      console.error(`changing outgoing multicast ${msg.newAddr}`);
+      sendSocket.setMulticastInterface(msg.newAddr);
+    }
+    sendSocket.send(
+      buf,
+      0,
+      buf.length,
+      PORTS[msg.mcast],
+      msg.mcast,
+      function(err) {
+        assert.ifError(err);
```

Comment:
If I force an EPERM error here with an iptables OUTPUT filter, the children processes don't get cleaned up. But, I don't think it needs to be addressed now, since this would show up as the first failure in a run it affects and is not different than [the test this is based on](https://github.com/nodejs/node/blob/master/test/internet/test-dgram-multicast-multi-process.js).